### ISSUE TITLE
Drop support for UMD modules

### ISSUE THREAD

**Issue by jonkoops** [Leaflet/Leaflet#8351]
### Checklist

- [X] I've searched through the [plugins](https://leafletjs.com/plugins.html) to make sure this feature isn't already available, or think it shouldn't require a plugin.
- [X] I've searched through the [current issues](https://github.com/Leaflet/Leaflet/issues) to make sure this feature hasn't been requested already.
- [X] I agree to follow the [Code of Conduct](https://github.com/Leaflet/Leaflet/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.

### Motivation

Supporting multiple module systems (e.g. CommonJS, AMD, UMD, etc) is becoming a maintenance burden and is complicating the project unnecessarily. The web is moving forwards towards standardized [JavaScript modules](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Modules) and Leaflet should take a stance to support this movement towards a simpler and more standardized method of distributing code.

Now that we have landed support for ES modules in 1.9 (to be released, see #8329) we should work towards shipping ESM as the one and only default option (with backwards compatibility for globals). This proposal removes the UMD build in favor of shipping only an ESM version.

### Suggested solution

dsasd

### Alternatives considered

dasd

**Comment by hooch** [comment]
Bring it on

**Comment by patrickarlt** [comment]
When could we expect a beta or release candidate for Leaflet 2.0? Speaking as a plugin maintainer it would be great to get a pretty long runway here to get plugins updated to new standard.

I would happy to create a "skeleton plugin" if we wanted to establish a good pattern for plugins to follow in terms of how to provide exports in different module formats, ect... and updating the [plugin guide](https://github.com/Leaflet/Leaflet/blob/main/PLUGIN-GUIDE.md)

**Comment by jonkoops** [comment]
@patrickarlt this all depends on our velocity in delivering the features we'd like to land. I hope to have some time soon to further develop the ESM and JavaScript classes support..

**Comment by CarlosTorrecillas** [comment]
Hello guys, first of all, thanks for the grat package! I wanted to ask you - with no pressure about it, obviously - if you made the decision about shipping ESM version of the package? Thanks a lot!

**Comment by jonkoops** [comment]
Yes, this issue will likely be closed with the release of version 2.0. We're all running this on volunteer work, so we cannot really give an eta of when this will be done.
